ext/dcos-installer/dcos_installer/action_lib/__init__.py

```diff
@@ -151,35 +163,38 @@ def _remove_host(state_file, host):
 
 
 @asyncio.coroutine
-def install_dcos(config, block=False, state_json_dir=None, hosts=[], async_delegate=None, try_remove_stale_dcos=False,
-                 roles=None, **kwargs):
-    if roles is None:
-        roles = ['master', 'agent']
+def install_dcos(config, block=False, state_json_dir=None, hosts=None, async_delegate=None, try_remove_stale_dcos=False,
```

Comment:
When this is called via a retry in async_server.py, hosts is passed through as a raw list of hosts, not a list of Node objects, which means later the `targets=` sometimes results in targets being set to a list of Node instances, and sometimes a list of whatever was in the json state.